utilities/unionTypeHelper.go

```diff
@@ -0,0 +1,135 @@
+package utilities
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type TypeHolder struct {
```

Comment:
i suggest this type Should be named as `UnionType` its Give more meaning. Let me know What you think.

Reply:
UnionType gives an impression that it is a combination of multiple types, which is not true in this case, this structure is a single type holder